mlflow_extensions/databricks/deploy/ez_deploy_ray_serve_entrypoint.py

```diff
@@ -0,0 +1,297 @@
+from vllm.utils import FlexibleArgumentParser
+from vllm.entrypoints.logger import RequestLogger
+
+
+from mlflow_extensions.databricks.deploy.ez_deploy import EzDeployConfig
+from utils import parse_vllm_configs,run_on_every_node
+from mlflow_extensions.databricks.deploy.gpu_configs import ALL_VALID_VM_CONFIGS
+
+# COMMAND ----------
+
+config = EzDeployConfig.from_json(ez_deploy_config)
+engine_process = config.to_proc()
+
+# COMMAND ----------
+
+node_info = [gpu for gpu in ALL_VALID_VM_CONFIGS if gpu.name == gpu_config['node_type_id']]
+assert len(node_info) == 1, f"Invalid gpu_config: {gpu_config}"
+
+if max_replica >1:
+
+  setup_ray_cluster(min_worker_nodes=min_replica, 
+                    max_worker_nodes=max_replica,
+                    num_cpus_head_node=4, 
+                    num_gpus_worker_node=node_info[0].gpu_count, 
+                    num_cpus_worker_node=node_info[0].cpu_count, 
+                    num_gpus_head_node=0)
+
+  # Pass any custom configuration to ray.init
+  ray.init(ignore_reinit_error=True)
+else:
+  # star local cluster
+  ray.init(include_dashboard=True ,ignore_reinit_error=True, dashboard_host = "0.0.0.0",dashboard_port= 8888)
+  display_databricks_driver_proxy_url(sc,8888, "ray-dashboard")
+# COMMAND ----------
+
+from mlflow.pyfunc import PythonModelContext
+artifacts = config.download_artifacts()
+ctx = PythonModelContext(artifacts=artifacts, model_config={})
+
+# COMMAND ----------
+
+@ray.remote(num_cpus=1)
+def download_model():
+    config.download_artifacts()
+
+
+# COMMAND ----------
+
+pg_resources ,parsed_args,engine_args = parse_vllm_configs(config,node_info,ctx)
+
+# COMMAND ----------
+
+_ = run_on_every_node(download_model)
+
+# COMMAND ----------
+
+app = FastAPI()
+
+
+@serve.deployment(
+    autoscaling_config={
+        "min_replicas": min_replica,
+        "max_replicas": max_replica,
+        "target_ongoing_requests": 10,
+    },
+    max_ongoing_requests=20,
+    ray_actor_options={"num_cpus": 4}
+)
+@serve.ingress(app)
+class VLLMDeployment:
+    def __init__(
+        self,
+        engine_args: AsyncEngineArgs,
+        model_config : EzDeployConfig,
+        response_role: str,
+        lora_modules: Optional[List[LoRAModulePath]] = None,
+        prompt_adapters: Optional[List[PromptAdapterPath]] = None,
+        request_logger: Optional[RequestLogger] = None,
+        chat_template: Optional[str] = None
+    ):
+        # self.artifacts = model_config.download_artifacts()
+        _ = run_on_every_node(download_model)
+        print(f"Starting with engine args: {engine_args}")
+        self.openai_serving_chat = None
+        self.engine_args = engine_args
+        self.response_role = response_role
+        self.lora_modules = lora_modules
+        self.prompt_adapters = prompt_adapters
+        self.request_logger = request_logger
+        self.chat_template = chat_template
+        self.engine = AsyncLLMEngine.from_engine_args(engine_args)
```

Comment:
in the next iteration we should abstract and move this into engine for VllmEngine so that we can have other ingress implementations.

Reply:
Yepp. I agree. I will add this to the issue's tracker and work on it.